packages/discordeno/src/bin/generate/typescript.ts

```diff
@@ -0,0 +1,193 @@
+import assert from 'node:assert'
+import { createWriteStream } from 'node:fs'
+import ts from 'typescript'
+
+// TODO: change this fileName to be based on the filesystem
+const fileName = process.argv.at(2)!
```

Comment:
1. I think it's better to throw an error if the user forgot to provide a file name, maybe something like this?
```suggestion
// TODO: change this fileName to be based on the filesystem
const fileName = process.argv.at(2)
if (!fileName) throw new Error("You need to provide a file name!")
```

2. What does "change this fileName to be based on the filesystem" mean?

Reply:
the script is not meant to be run manually, the code will attempt to get the .d.ts file from the node_modules (currently it uses a .ts so that needs work too)